src/Common/src/System/Globalization/IdnMapping.Unix.cs

```diff
@@ -11,44 +11,37 @@ private unsafe string GetAsciiCore(char* unicode, int count)
             uint flags = Flags;
             CheckInvalidIdnCharacters(unicode, count, flags, nameof(unicode));
 
+            int length = Interop.GlobalizationNative.ToAscii(flags, unicode, count, null, 0);
+            if (length == 0)
+            {
+                throw new ArgumentException(SR.Argument_IdnIllegalName, nameof(unicode));
+            }
+
             const int StackAllocThreshold = 512;
-            if (count < StackAllocThreshold)
+            if (length < StackAllocThreshold)
             {
-                char* output = stackalloc char[count];
-                return GetAsciiCore(unicode, count, flags, output, count, reattempt: true);
+                char* output = stackalloc char[length];
+                return GetAsciiCore(unicode, count, flags, output, length);
             }
             else
             {
-                char[] output = new char[count];
+                char[] output = new char[length];
                 fixed (char* pOutput = output)
                 {
-                    return GetAsciiCore(unicode, count, flags, pOutput, count, reattempt: true);
+                    return GetAsciiCore(unicode, count, flags, pOutput, length);
                 }
             }
         }
 
-        private unsafe string GetAsciiCore(char* unicode, int count, uint flags, char* output, int outputLength, bool reattempt)
+        private unsafe string GetAsciiCore(char* unicode, int count, uint flags, char* output, int outputLength)
```

Comment:
I'm not sure we can, as this method is called with a stackalloced char* and a fixed char* in different scopes so this avoids code duplication